### PR TITLE
feat: add passive sources to projections

### PR DESCRIPTION
**Pull request by axejintao**
# Summary

- Include baseline APR in harvest and yield returns

### CODE REVIEW

File: src/vaults/vaults.service.ts [hunk 1]
```diff
@@ -170,16 +194,21 @@ export class VaultsService {
...
     // refactor regV2, Data Objects interfaces should be in api alongside with models
-    vault: Pick<VaultDTO, 'value' | 'balance' | 'available' | 'lastHarvest'>,
+    vault: Pick<VaultDTO, 'value' | 'balance' | 'available' | 'lastHarvest' | 'underlyingToken'>,
     pendingHarvest: VaultPendingHarvestData,
   ): VaultYieldProjection {
     const { value, balance, available, lastHarvest } = vault;
+
     const harvestValue = pendingHarvest.harvestTokens.reduce((total, token) => (total += token.value), 0);
     const yieldValue = pendingHarvest.yieldTokens.reduce((total, token) => (total += token.value), 0);
+    const harvestCompoundValue = pendingHarvest.harvestTokens
```
Comment (axejintao) [inline src/vaults/vaults.service.ts:204]: first change here - identify harvest components that are intended for compounding 

this gives us a better read on the true "compounding portion" for apy calculation

File: src/vaults/vaults.service.ts [hunk 2]
```diff
@@ -215,17 +244,18 @@ export class VaultsService {
...
+    compoundingValue = 0,
   ): number {
     if (lastHarvested === 0 || value === 0 || pendingValue === 0) {
       return 0;
     }
     const duration = Date.now() / 1000 - lastHarvested;
-    const apr = (pendingValue / value) * (ONE_YEAR_SECONDS / duration);
-    if (!apy) {
-      return apr * 100;
+    const apr = (pendingValue / value) * (ONE_YEAR_SECONDS / duration) * 100;
```
Comment (axejintao) [inline src/vaults/vaults.service.ts:253]: this compounding value is then passed through here to either use to calculate apy or to treat just as apr
in other words, we would use that amount that gets compounded and calculate the apy instead

File: src/rewards/rewards.utils.ts [hunk 3]
```diff
@@ -35,14 +35,6 @@ export async function getTreeDistribution(chain: Chain): Promise<RewardMerkleDis
   }
 }
 
-export function noRewards(VaultDefinition: VaultDefinition, token: Token) {
```
Comment (axejintao) [inline src/rewards/rewards.utils.ts:38]: not used anywhere, no reason to even include 'no rewards'

File: src/vaults/vaults.service.ts [hunk 4]
```diff
@@ -144,7 +144,31 @@ export class VaultsService {
     vault.apy = vault.sourcesApy.map((s) => s.apr).reduce((total, apr) => (total += apr), 0);
     vault.protocol = vaultDefinition.protocol ?? Protocol.Badger;
     vault.lastHarvest = pendingHarvest.lastHarvestedAt;
-    vault.yieldProjection = this.getVaultYieldProjection(vault, pendingHarvest);
+
+    const harvestProjection = this.getVaultYieldProjection(vault, pendingHarvest);
+    const passiveSources = sourcesApr.filter((s) => s.type === SourceType.TradeFee || s.type === SourceType.Emission);
```
Comment (axejintao) [inline src/vaults/vaults.service.ts:149]: once the projections based on the harvest values are calculated, 

we can then grab the sources which are passive and considered to always be active to enrich the data returned by harvest / yield estimates

this is intended to help the projection values properly return the estimates they should be giving back 

NOTE: in the case of the badger rewards, this is the pro rata APR 

so while its useful for some general idea of what is going on the UI for example should not utilize this metric and instead favor the sources objects as a source of truth to derive an individual user's level of compensation wrt emitted badger rewards

File: src/vaults/vaults.service.ts [hunk 5]
```diff
@@ -144,7 +144,31 @@ export class VaultsService {
...
+    const passiveSourcesApr = passiveSources.reduce((total, s) => (total += s.apr), 0);
+    const convertedPassiveSources = passiveSources.map((s) => {
```
Comment (axejintao) [inline src/vaults/vaults.service.ts:151]: coerce passive sources into yield token objects

**Review by mrbasado — APPROVED** [review]

**Pull request merged**